src/main/java/uk/gov/pay/ledger/transaction/service/TransactionMetadataService.java

```diff
@@ -49,8 +49,9 @@ public void upsertMetadataFor(Event event) {
                         Iterator<String> metadataKeys = eventDataNode.get("external_metadata").fieldNames();
```

Comment:
Would it be slightly cleaner to change this to iterate over the `fields()` of the JsonNode so we don't have to get the field by the key again when we iterate?

Reply:
I have considered this approach but if we iterate over `fields()` then we still need to get the relevant key for the upsert.

Reply:
This is what I was thinking:

```
                        eventDataNode.get("external_metadata").fields().forEachRemaining(metadata -> {
                            metadataKeyDao.insertIfNotExist(metadata.getKey());
                            String value = metadata.getValue().textValue();
                            transactionMetadataDao
                                    .upsert(transactionEntity.getId(), metadata.getKey(), value);
                        });
```